specs/experimental/builder.md

```diff
@@ -0,0 +1,172 @@
+# Builder API
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Builder API](#builder-api)
+  - [Overview](#overview)
+  - [Sequencer Interaction](#sequencer-interaction)
+    - [Requesting a Block](#requesting-a-block)
+      - [Liveness Failsafe](#liveness-failsafe)
+    - [Mempool Forwarding](#mempool-forwarding)
+    - [Builder Configuration](#builder-configuration)
+  - [Structures](#structures)
+    - [`PayloadRequestV1`](#payloadrequestv1)
+    - [`BuilderPayloadV1`](#builderpayloadv1)
+  - [Methods](#methods)
+    - [`builder_getPayloadV1`](#builder_getpayloadv1)
+      - [Request](#request)
+      - [Response](#response)
+      - [Specification](#specification)
+    - [`builder_forwardTransactionV1`](#builder_forwardtransactionv1)
+      - [Request](#request-1)
+      - [Response](#response-1)
+      - [Specification](#specification-1)
+  - [Alternative Approach for Review (!!!!Most Likely To Be Deleted but Keeping for Consideration!!!!)](#alternative-approach-for-review-most-likely-to-be-deleted-but-keeping-for-consideration)
+    - [`engine_forkchoiceUpdatedV4`](#engine_forkchoiceupdatedv4)
+    - [Extended `PayloadAttributesV4`](#extended-payloadattributesv4)
+      - [Process](#process)
+    - [Rationale](#rationale)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+This document provides a in-depth specification for integrating a Builder API within the Optimism Protocol and Stack. The Builder API provides a standardized interface for block construction and transaction management between the Sequencer and an external network of Block Builders. The specified interactions are the minimum viable design needed to allow an external Builder Network enabling future protocol experimentation and a path towards Sequencer decentralization. Additionally this minimum viable design includes a fallback as a training wheel to ensure liveness and network performance. This document ***does not*** outline how to safely enable a permissionless Builder Network to serve payloads or how to manage these configs on L1.
```

Comment:
> provides a standardized interface for block construction and transaction management between the Sequencer and an external network of Block Builders

I think this spec is broad enough that I would challenge narrowing it to an "external network of Block Builders". Reading over the rest of the doc, it occurs to me that this builder distinction would also be satisfied by a local builder. Rather than assume that the builder component lives elsewhere, the builder component *could* just be library or local package that satisfies the block building API, right?

I find the modularity of this proposal to very compelling, and I don't want to constrain the nature of where a block builder might live, or how it might function.

Reply:
good point, changed to "local Block Builder"